tutorials/tools/binary-to-decimal.md

```diff
@@ -0,0 +1,146 @@
+A [binary number](https://en.wikipedia.org/wiki/Binary_number) is a number expressed in base-2, as opposed to conventional decimal numbers in base-10.
+
+Below is a live calculator.
+
+<style>
+  #binary {
+    font-size: 1.5em;
+    padding: 0.25em;
+    border-radius: 4px;
+  }
+
+  #result {
+    display: inline-block;
+    font-size: 1.5em;
+    padding: 0.25em;
+    border-radius: 4px;
+    background-color: #ddd;
+    width: 310px;
+    overflow-wrap: anywhere;
+  }
+
+  #right-arrow {
+    font-size: 2em;
+    padding: 0.25em;
+    padding-left: 0.5em;
+    padding-right: 0.5em;
+  }
+
+  @media (max-width: 1000px) {
+    #decimal {
+      width: 100%;
+    }
+
+    #result {
+      width: 8em;
+    }
+  }
+</style>
+
+<table>
+  <thead>
+    <th>Binary</th>
+    <th></th>
+    <th>Decimal</th>
+  </thead>
+  <tr>
+    <td>
+      <input type="number" id="binary" placeholder="101010" />
+    </td>
+    <td id="right-arrow">
+      &#10140;
+    </td>
+    <td>
+      <div id="result"></div>
+    </td>
+  </tr>
+</table>
+<script>
+  document.querySelector('#result').innerHTML = parseInt(101010, 2);
+  document.querySelector('#binary').addEventListener('change', function() {
+    const value = parseInt(document.querySelector('#binary').value, 2);
+    if (value == null || isNaN(value)) {
+      return;
+    } 
+    document.querySelector('#result').innerHTML = value;
+  });
+  document.querySelector('#binary').addEventListener('keyup', function() {
+    const value = parseInt(document.querySelector('#binary').value, 2);
+    if (value == null || isNaN(value)) {
+      return;
+    } 
+    document.querySelector('#result').innerHTML = value;
+  });
+  document.querySelector('#binary').focus();
+</script>
+
+## How the Calculator Works
+
+Converting binary numbers to decimal is easy.
+For example, `let x = 101010` creates a new variable `x` that contains the number `101010`.
+JavaScript has a [`parseInt()` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt) that takes a `binary` and `radix` parameter, and returns a number.
+Calling `parseInt(binary, radix)` tells JavaScript to convert `binary` to a number containing the decimal representation of `101010`.
+If `binary` is not a string, it will be converted to one using the `toString()` function.
+
+```javascript
+let x = 101010;
+
+parseInt(x, 2) // 42
+```
+
+The `toString()` method also handles non-integers and negative numbers. For example:
+
+```javascript
+x = -101010
+parseInt(x, 2); // -42
+
+x = 101010.101010; // 42
+```
+
+## Binary Numeric Literals
+
+You can also use binary numeric numerals to auto convert it in your code.
+
+```javascript
+x = 0b101010;
```

Comment:
Let's get rid of this section, not necessary. Converting literals is not important for this tutorial.